osu.Game/Screens/OnlinePlay/Multiplayer/Spectate/MultiSpectatorScreen.cs

```diff
@@ -178,19 +178,45 @@ protected override void Update()
         {
             base.Update();
 
-            if (!isCandidateAudioSource(currentAudioSource?.SpectatorPlayerClock))
-            {
-                currentAudioSource = instances.Where(i => isCandidateAudioSource(i.SpectatorPlayerClock)).MinBy(i => Math.Abs(i.SpectatorPlayerClock.CurrentTime - syncManager.CurrentMasterTime));
+            var maximizedPlayer = getMaximizedPlayer();
+            PlayerArea? newAudioSource = null;
```

Comment:
please rename to `candidateAudioSource`, being called `newAudioSource` makes this confusing because the switch of the audio source doesn't actually happen before the `newAudioSource != currentAudioSource` conditional below